main.go

```diff
@@ -88,9 +88,24 @@ func main() {
 		exit(1)
 	}
 
-	adds, deletes := localRecords.Diff(records)
+	// Cast the records from Cloudflare to a recordCollection.
+	existingRecords := recordCollection(records)
 
-	numChanges := len(adds) + len(deletes)
+	// Find records only present at cloudflare - and records only present in
+	// the local zone. This will be the basis for the add/delete collections.
+	addCandidates := fileRecords.Difference(existingRecords, FullMatch)
+	deleteCandidates := existingRecords.Difference(fileRecords, FullMatch)
+
+	// If we find the intersection between local and remote, we should have a
```

Comment:
The mention of "local" and "remote" here doesn't match the current variable naming in the code.